system-overview/cryptography.md

```diff
@@ -0,0 +1,115 @@
+other private keys which unlock a wallet that corresponds to your ship's
+[proxies](@/docs/glossary/proxies.md). We refer collectively to these keys as
+your _Azimuth keys_.
+
+Only [planets](@/docs/glossary/planet.md), [stars](@/docs/glossary/star.md), and
+[galaxies](@/docs/glossary/galaxy.md) have Azimuth keys.
+[Moons](@/docs/glossary/moon.md) and [comets](@/docs/glossary/comet.md) do not,
+as they do not utilize the Ethereum blockchain.
+
+It is important to note that no Azimuth keys are stored anywhere within your
+ship's [pier](@/docs/glossary/pier.md) - Ethereum and Urbit ID are entirely
+separate entities from Urbit itself, and so you lose access to your Azimuth
+private keys there is no way to retrieve them somehow from your ship.
+
+For more information on the usage of these keys and the associated proxies, see
+the [Azimuth documentation](@/docs/azimuth/azimuth.md).
+
+#### Networking keys
+
+All communications in Urbit over the [Ames](@/docs/glossary/ames.md) networking
+are end-to-end encrypted, and thus your ship stores its own public/private pair
+of _networking keys_ utilized for encryption and authentication. Networking keys
+for all ship types are stored within the ship's Jael vane.
+
+For planets, stars, and galaxies, your private networking key is a
+necessary input (the [keyfile](@/docs/glossary/keyfile.md)) for the initial boot
+sequence of your ship (sometime called its `%dawn`). The associated public key
+is then uploaded to the Ethereum blockchain using one of your Azimuth private
+keys - either the one associated to the wallet which owns the ship, or the one
+which holds the management proxy.
+
+For moons, the private networking key is generated by the parent ship and
+injected into the `%dawn` sequence for the moon, and its public key is stored by
+the parent in its Jael vane. Moon networking keys may be changed by the parent
+ship, but not by the moon itself.
+
+For comets, their 128-bit `@p` name is actually their public key, and the
+"mining" process to generate a comet consists of guessing a private key for
+which the associated public key has a galaxy that has already been pre-assigned
+to assist its peer discovery process. Thus, comets cannot change their
+networking keys - to get a new private networking key, a new comet must be
+generated. A comet utilizes its private networking key stored in Jael to sign an
+unencrypted attestation packet to verify that it is the owner of the associated
+public key. Because of this, it is currently impossible for a ship to initiate
+communication with a comet - the comet must always be the initiator. This means
+that ultimately two comets cannot communicate with one another unless they have
+somehow verified each others' public keys via some other method. This is merely
+a technical limitation imposed by the design of the system, not an intentional
+handicapping of comet abilities. It is expected that this limitation will
+someday be lifted.
```

Comment:
Worth mentioning explicitly that the attestation packet is part of the initial handshake, to make more clear why it would block communication in a specific direction.

Also paging @belisarius222: wasn't a dumb quick fix for this in the pipeline already? (And if not, friendly reminder that people keep hitting this case!)